rollout/bluegreen.go

```diff
@@ -61,52 +62,52 @@ func (c *RolloutController) rolloutBlueGreen(r *v1alpha1.Rollout, rsList []*apps
 
 	if scaledUp {
 		logCtx.Infof("Not finished reconciling new ReplicaSet '%s'", newRS.Name)
-		return c.syncRolloutStatusBlueGreen(previewSvc, activeSvc, roCtx, false)
+		return c.syncRolloutStatusBlueGreen(previewSvc, activeSvc, roCtx)
 	}
 
 	if scaledDown {
 		logCtx.Info("Not finished reconciling old replica sets")
-		return c.syncRolloutStatusBlueGreen(previewSvc, activeSvc, roCtx, false)
+		return c.syncRolloutStatusBlueGreen(previewSvc, activeSvc, roCtx)
 	}
 	if switchPreviewSvc {
 		logCtx.Infof("Not finished reconciling preview service' %s'", previewSvc.Name)
-		return c.syncRolloutStatusBlueGreen(previewSvc, activeSvc, roCtx, false)
+		return c.syncRolloutStatusBlueGreen(previewSvc, activeSvc, roCtx)
 	}
 
 	if !replicasetutil.ReadyForPause(r, newRS, allRSs) {
 		logutil.WithRollout(r).Infof("New RS '%s' is not fully saturated", newRS.Name)
-		return c.syncRolloutStatusBlueGreen(previewSvc, activeSvc, roCtx, false)
+		return c.syncRolloutStatusBlueGreen(previewSvc, activeSvc, roCtx)
 	}
 
 	noFastRollback := true
 	if newRS != nil {
-		_, ok := newRS.Annotations[v1alpha1.DefaultReplicaSetScaleDownDeadlineAnnotationKey]
-		noFastRollback = !ok
-		if !noFastRollback {
-			logCtx.Infof("Detected '%s' annotation and will skip pause", newRS.Name)
+		_, hasScaleDownDeadlineAnnotationKey := newRS.Annotations[v1alpha1.DefaultReplicaSetScaleDownDeadlineAnnotationKey]
+		if hasScaleDownDeadlineAnnotationKey {
+			logCtx.Infof("Detected scale down annotation for ReplicaSet '%s' and will skip pause", newRS.Name)
 		}
+		noFastRollback = !hasScaleDownDeadlineAnnotationKey
```

Comment:
Can we make this a little clearer and explicit:
```golang
if hasScaleDownDeadlineAnnotationKey {
    logCtx.Infof("Detected scale down annotation for ReplicaSet '%s' and will skip pause", newRS.Name)
} else {
    noFastRollback = false
}
```

Reply:
👍

Reply:
Actually, I think the logic should be ```
if hasScaleDownDeadlineAnnotationKey {
    logCtx.Infof("Detected scale down annotation for ReplicaSet '%s' and will skip pause", newRS.Name)
    noFastRollback = false
}``` but I think this is proving the point that it can be clearer and explicit